### PR TITLE
Add emoji to test template values to test we can deal with unicode

### PR DESCRIPTION
**Pull request by aharpervc**
This actually does not work correctly at the moment. Consul UI shows the correct value, but the emoji are being templated out as `??` (the emoji from the file templates come out fine).

image

### CODE REVIEW

**Comment by aharpervc** [comment]
Legit failure, which is why I am wanting to add this test: https://circleci.com/gh/veracross/consult/8

**Comment by aharpervc** [comment]
Fixed: https://circleci.com/gh/veracross/consult/11

File: lib/consult/template.rb [hunk 1]
```diff
@@ -75,7 +75,7 @@ def contents
 
     def consul_contents(location)
       [@config[location]].compact.flatten.map do |key|
-        Diplomat::Kv.get(key, options: nil, not_found: :return, found: :return)
+        Diplomat::Kv.get(key, options: nil, not_found: :return, found: :return).force_encoding 'utf-8'
```
Comment (sixfeetover) [inline lib/consult/template.rb:78]: What encoding are you seeing this come back as without this? And did you try just `encode` instead of `force`?

**Review by mmartell37 — APPROVED** [review]
Notwithstanding @sixfeetover's comment, this LGTM 🌎

File: lib/consult/template.rb (hunk 1 above)
Comment (aharpervc) [reply lib/consult/template.rb:78]: I updated the commit message to clarify some more, but it's `ASCII-8BIT`. That's probably because of `Base64.decode64`. The problem is, if you have a unicode string in Consul, then it's escaped when it comes out here, which normally wouldn't be a big deal but Ruby throws an exception when trying to combine these with the UTF-8 strings from files (explicitly set below in `disk_contents`).

**Comment by aharpervc** [comment]
Jeff brought up the question of what should be done if there's an invalid/undecodable string in Consul. Presumably that would have been an exception before, and still would be if the string can't be converted.

**Review by sixfeetover — APPROVED** [review]
Confirmed that `decode64` is returning `ASCII-8BIT` encoding, so yes, I'm good with this change to ensure matching encodings upon merging the template.

Unclear whether we want to explicitly handle encoding conversions errors - it's possible we want that to fail rather than cover it up, so I'm ok not accounting for that in this PR.
